### PR TITLE
feat(scheduler): Create Interval DTO and Model

### LINKED ISSUE #426: [V2 schedular] Create Interval DTO and Model
Create Interval DTO and Model for V2 API

### PR DESCRIPTION
**Pull request by weichou1229**
Close #426

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the new behavior?
Create Interval DTO and Model for scheduler V2 API

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?
- I remove the Cron field from Interval because it does nothing in the scheduler service, we can add it back if we really need it.
- Reuse the validation tag to verify the Frequency field

## Other information

### CODE REVIEW

File: v2/validator.go [hunk 1]
```diff
@@ -21,16 +21,18 @@ import (
 var val *validator.Validate
 
 const (
-	autoEventFrequencyTag       = "edgex-dto-autoevent-frequency"
+	dtoFrequencyTag             = "edgex-dto-frequency"
```
Comment (lenny-intel) [inline v2/validator.go:24]: This change impacts AutoEvent DTO correct. Updates needed there?

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: v2/dtos/requests/interval_test.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func addIntervalRequestDate() AddIntervalRequest {
```
[the remark this answers is not in the document]
Comment (weichou1229) [reply v2/dtos/requests/interval_test.go]: Fixed the typo.

File: v2/validator.go (hunk 1 above)
Comment (weichou1229) [reply v2/validator.go:24]: Since the AutoEvent and Interval both have a frequency field, so I rename the tag to make it more general for reusing.

Comment (lenny-intel) [reply v2/validator.go:24]: I understand and agree with the name change, but expected to see changes to the AutoEvent DTO to use new name.

**Review by lenny-intel — CHANGES_REQUESTED** [review]
Note that this PR should be on-hold per Core WG decision to revisit the Support service API design before moving forward with implementations.

File: v2/validator.go (hunk 1 above)
Comment (weichou1229) [reply v2/validator.go:24]: No problem, change made there https://github.com/edgexfoundry/go-mod-core-contracts/pull/427/files#diff-fbe92df0cf3b16ad6c0f28c3528270b3310d83530f06fe639ca331a0096ece02R14

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]
add new request factory method

File: v2/validator.go (hunk 1 above)
Comment (lenny-intel) [reply v2/validator.go:24]: Thanks!

**Comment by weichou1229** [comment]
Create request factory method via PR https://github.com/edgexfoundry/go-mod-core-contracts/pull/462

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Comment by cloudxxx8** [comment]
@weichou1229 please fix the unit test

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Comment by lenny-intel** [comment]
@weichou1229 , PR needs to be rebased to be brought up to date with the recent V2 Go Model changes in your imports.

**Comment by weichou1229** [comment]
Modified import path for V2 Go Model changes.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**